codice/ddf
# DDF-5260 Updates editing restrictions to take into account write access

#### What does this PR do?
 - Updates the edit restriction checking to check both the admin config and write access for the result.
 - Updates the usage of these restrictions to be local and removes the global 'is-editing-restricted' class that was placed on the body.  This lets the choice be decided on a per result basis and makes it so components who want to know are in charge of figuring it out.

#### How should this be tested?
Verify that the admin config for restricting editing in general still works as intended.

Verify that without that config on, restricting a user to only have read access to a metacard has a similar effect.

In both, when a restricted metacard is included in a bulk selection, the inspector is limited to the details tab without an edit footer.  It also removes from the actions dropdown adding and removing of attributes.

In a single selection, the user can reach summary, details, history, associations, quality, etc but can't perform edit actions in any of them.

#### What are the relevant tickets?
Fixes: #5260
